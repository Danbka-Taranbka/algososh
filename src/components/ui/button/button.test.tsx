import renderer from 'react-test-renderer';
import { Button } from './button'
import { render, screen, fireEvent } from "@testing-library/react";

describe('Button', () => {

  it('is rendered on page!', () => {
    render(<Button/>);
    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();
  });

  it('renders correctly without text!', () => {
    const tree = renderer.create(<Button/>);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with text!', () => {
    const tree = renderer.create(<Button text='Рассчитать'/>);
    expect(tree).toMatchSnapshot();
  });

  it('disabled renders correctly!', () => {
    const tree = renderer.create(<Button disabled={true}/>);
    expect(tree).toMatchSnapshot();
  });

  it('loading renders correctly!', () => {
    const tree = renderer.create(<Button isLoader={true}/>);
    expect(tree).toMatchSnapshot();
  });
  
  it('callback works correctly!', () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick} />);
    fireEvent.click(screen.getByRole("button"));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});