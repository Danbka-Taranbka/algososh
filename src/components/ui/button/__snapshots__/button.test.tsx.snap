// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button disabled renders correctly! 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button loading renders correctly! 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button renders correctly with text! 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  name="Рассчитать"
  type="button"
>
  <p
    className="text undefined"
  >
    Рассчитать
  </p>
</button>
`;

exports[`Button renders correctly without text! 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
